pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -0,0 +1,1577 @@
+
+			ensure!(n_currencies <= max_currencies, Error::<T>::CurrencyCount);
+
+			// refunding can only be triggered on a live pool
+			ensure!(pool_details.state.is_live(), Error::<T>::PoolNotLive);
+
+			if let Some(caller) = maybe_check_manager {
+				// TODO: should the owner be authorized as well?
+				ensure!(pool_details.is_manager(caller), Error::<T>::NoPermission);
+			}
+
+			let total_collateral_issuance =
+				T::CollateralCurrencies::total_balance(pool_details.collateral_id.clone(), &pool_id.clone().into());
+			// nothing to distribute
+			ensure!(
+				total_collateral_issuance > CollateralCurrenciesBalanceOf::<T>::zero(),
+				Error::<T>::NothingToRefund
+			);
+
+			let has_holders = pool_details
+				.bonded_currencies
+				.iter()
+				.any(|asset_id| T::Fungibles::total_issuance(asset_id.clone()) > FungiblesBalanceOf::<T>::zero());
+			// no token holders to refund
+			ensure!(has_holders, Error::<T>::NothingToRefund);
+
+			// switch pool state to refunding
+			let mut new_pool_details = pool_details;
+			new_pool_details.state.start_refund();
+			Pools::<T>::set(&pool_id, Some(new_pool_details));
+
+			Self::deposit_event(Event::RefundingStarted { id: pool_id });
+
+			Ok(n_currencies)
+		}
+
+		/// Starts the destruction process for a pool.
+		/// This function is called by both the normal and forced start_destroy
+		/// functions.
+		///
+		/// # Parameters
+		/// - `pool_id`: The identifier of the pool to be destroyed.
+		/// - `max_currencies`: The currency count in the pool, required for
+		///   weight calculation.
+		/// - `force_skip_refund`: A flag to skip the refund check.
+		/// - `maybe_check_manager`: An optional parameter to check if the
+		///   caller is a manager.
+		///
+		/// # Returns
+		/// - `Result<u32, DispatchError>`: The number of currencies in the pool
+		///   or an error.
+		///
+		/// # Errors
+		/// - `Error::<T>::PoolUnknown`: If the pool does not exist.
+		/// - `Error::<T>::CurrencyCount`: If the number of currencies exceeds
+		///   `max_currencies`.
+		/// - `Error::<T>::PoolNotLive`: If the pool is not in a live or
+		///   refunding state.
+		/// - `Error::<T>::NoPermission`: If the caller is not the owner or
+		///   manager of the pool.
+		/// - `Error::<T>::LivePool`: If there are holders or collateral to
+		///   distribute.
+		fn do_start_destroy_pool(
+			pool_id: T::PoolId,
+			max_currencies: u32,
+			force_skip_refund: bool,
+			maybe_check_manager: Option<&AccountIdOf<T>>,
+		) -> Result<u32, DispatchError> {
+			let pool_details = Pools::<T>::get(&pool_id).ok_or(Error::<T>::PoolUnknown)?;
+
+			let n_currencies = Self::get_currencies_number(&pool_details);
+
+			ensure!(n_currencies <= max_currencies, Error::<T>::CurrencyCount);
+
+			ensure!(
+				pool_details.state.is_live() || pool_details.state.is_refunding(),
+				Error::<T>::PoolNotLive
+			);
+
+			if let Some(caller) = maybe_check_manager {
+				// TODO: should this be permissionless if the pool is in refunding state?
```

Comment:
Do we still want to tackle this?

Reply:
I'd keep TODOs where the desired behaviour is not entirely clear, expecting that these can be resolved best in the trial phase (once deployed on a testnet)